lib/string_reverse.rb

```diff
@@ -1,4 +1,18 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+  unless my_string == nil || my_string.empty? || my_string.length == 1
+
+    index = 0
+    last = my_string.length - 1
+
+    while index <= last
+      temp = my_string[index]
+      my_string[index] = my_string[last]
+      my_string[last] = temp
+      index += 1
+      last -= 1
+    end
+  end
+
+  return my_string

```

Comment:
Since you're reversing the string in place, you don't need to return the string.